lib/Doctrine/DBAL/DBALException.php

```diff
@@ -28,21 +22,28 @@ class DBALException extends Exception
      */
     public static function driverExceptionDuringQuery(Driver $driver, Throwable $driverEx, $sql, array $params = [])
     {
-        $msg = "An exception occurred while executing '" . $sql . "'";
-        if ($params) {
-            $msg .= ' with params ' . self::formatParameters($params);
-        }
-        $msg .= ":\n\n" . $driverEx->getMessage();
+        $messageFormat = <<<'MESSAGE'
+An exception occurred while executing "%s"%s:
+
+%s.
```

Comment:
The driver exception will likely have a period in the end. In a worst-case scenario, it's better to not have a period at all than have two.